### PR TITLE
build: use Codecov GitHub Action

### PR DESCRIPTION
**Pull request by xcoulon**
Paired with https://github.com/openshift/release/pull/19491

Fixes https://issues.redhat.com/browse/CRT-1033

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

**Review by MatousJobanek — APPROVED** [review]

File: .github/workflows/ci-build.yml [hunk 1]
```diff
@@ -10,6 +10,34 @@ on:
       - master
 
 jobs:
+  test-with-coverage:
+    name: Test with Coverage
+    runs-on: ubuntu-latest
```
Comment (alexeykazakov) [inline .github/workflows/ci-build.yml:15]: As far as I remember we had issues with some dependencies in the latest Ubuntu and had to switch to some specific version. So maybe it's safer to stick with some LTS version (*.04 versions)? cc: @MatousJobanek

**Review by alexeykazakov — APPROVED** [review]

File: .github/workflows/ci-build.yml (hunk 1 above)
Comment (xcoulon) [reply .github/workflows/ci-build.yml:15]: oh, I did not know about that 😮 
yes, indeed, we could use Ubuntu 18.04 or 20.04: https://github.com/actions/virtual-environments#available-environments

Comment (xcoulon) [reply .github/workflows/ci-build.yml]: switched to Ubuntu 20.04 (https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md) in b8109ce

Comment (alexeykazakov) [reply .github/workflows/ci-build.yml]: Yeah. 20.04 would be preferable: https://ubuntu.com/about/release-cycle

Comment (MatousJobanek) [reply .github/workflows/ci-build.yml]: it's not a big deal for these CI jobs - there was some problem with a specific version of Podman that was installed in the latest version of Ubuntu - that's why we use 18.04 in CD job

Comment (MatousJobanek) [reply .github/workflows/ci-build.yml]: apart from that - I guess that 20.04 & latest is the same: https://github.com/actions/virtual-environments#available-environments

Comment (xcoulon) [reply .github/workflows/ci-build.yml]: right, `20.04` and `latest` are equivalent. I'll go ahead and merge with `20.04` for now, but we can always change back to `latest` (or something else) in the future if needed

**Pull request merged**